tools/toolchains/iar.py

```diff
@@ -53,6 +52,15 @@ def __init__(self, target, notify=None, macros=None, build_profile=None,
             build_dir=build_dir,
             build_profile=build_profile
         )
+        if  getattr(target, "default_lib", "std") == "small":
+            if (
+                hasattr(target, "supported_c_libs") == False
+                or "iar" not in target.supported_c_libs
+                or "small" not in target.supported_c_libs["iar"]
+            ):
+                raise NotSupportedException(
+                    "target.default_lib small not supported for this target."
```

Comment:
Write "small C library not supported for this target" so you don't have to change the message if/when default_lib becomes "c_lib".

Reply:
The changes for gcc.py are missing? (was it tested)